src/SW_Main_lib.c

```diff
@@ -240,26 +244,32 @@ void sw_write_logs(Bool QuietMode, LOG_INFO* LogInfo) {
 	int warnMsgNum, warningUpperBound = LogInfo->numWarnings;
 	Bool tooManyWarns = swFALSE;
 	char tooManyWarnsStr[MAX_LOG_SIZE];
+    char tooManyStrFmt[] = "There were a total of %d warnings and"\
+                          " only %d were printed.\n";
 
 	if(warningUpperBound > MAX_MSGS) {
 		warningUpperBound = MAX_MSGS;
 		tooManyWarns = swTRUE;
 
-        sprintf(tooManyWarnsStr, "There were a total of %d warnings and"\
-                                 " only %d were printed.\n",
-                                 LogInfo->numWarnings, MAX_MSGS);
+        snprintf(tooManyWarnsStr, sizeof(tooManyStrFmt), tooManyStrFmt,

```

Comment:
`snprintf(tooManyWarnsStr, sizeof(tooManyStrFmt), ...` doesn't do the correct thing. This needs to be `snprintf(tooManyWarnsStr, sizeof(tooManyWarnsStr), ...` instead.

For instance, 
```
snprintf(tooManyWarnsStr, sizeof(tooManyStrFmt), tooManyStrFmt, 1000000, MAX_MSGS);
```
will cut off several of the last characters of `tooManyWarnsStr` - this is because the values get injected after `sizeof(tooManyStrFmt)` is determined.